ttnn/cpp/ttnn/tensor/layout/tensor_layout.hpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+
+#include "alignment.hpp"
+#include "size.hpp"
+#include "page_config.hpp"
+
+#include "ttnn/tensor/enum_types.hpp"
+#include "ttnn/tensor/types.hpp"
+#include "ttnn/tensor/shape/shape.hpp"
+
+namespace tt::tt_metal {
+
+using Strides = std::vector<size_t>;
+
+// TensorLayout describes how a tensor is laid out in memory
+// It takes datatype, layout (eg. TILE vs. RM), memory (eg. DRAM vs. L1), sharding (ie. how you want to cut your logical shape)
+// And provides information required to physically lay out the tensor in memory
+class TensorLayout {
+public:
+    TensorLayout(DataType dtype, const PageConfig& page_config, const MemoryConfig& memory_config);
+
+    // This is a static method instead of a constructor to make it easy to find and remove all of its usages in the codebase.
+    [[deprecated("Use of LegacyPaddedShape is deprecated. Please use constructor with Alignment instead.")]]
```

Comment:
This error message says to "use constructor with Alignment", but there is no such constructor

Reply:
Corrected, thanks